### ISSUE TITLE
New functions for select

### ISSUE THREAD

**Issue by Timu57** [rakibtg/SleekDB#170]
#169 brought the idea.

We could implement functions for select (like we have with select in conjunction with groupBy).

# [List of all Mysql-Functions](https://www.w3schools.com/sql/sql_ref_mysql.asp)

I think the following functions could be useful for us? Would like to hear your opinions!

### CONCAT
https://www.w3schools.com/sql/func_mysql_concat_ws.asp

> Alters single document

```sql
SELECT CONCAT_WS(" ", Address, PostalCode, City) AS Address
```

### LENGTH
https://www.w3schools.com/sql/func_mysql_length.asp

> Alters single document

```sql
SELECT LENGTH(CustomerName) AS LengthOfName
```

### LOWER
https://www.w3schools.com/sql/func_mysql_lower.asp

> Alters single document

```sql
SELECT LOWER(CustomerName) AS LowercaseCustomerName
```

### POSITION
https://www.w3schools.com/sql/func_mysql_position.asp

> Alters single document

OR `LOCATE` OR `INSTR`
They do all the same thing, but I am not sure what's the best naming.

```sql
SELECT POSITION("a" IN CustomerName)
```

### REPEAT

https://www.w3schools.com/sql/func_mysql_repeat.asp

> Alters single document

This method will not be implemented.

```sql
SELECT REPEAT(CustomerName, 2)
```

### UPPER
https://www.w3schools.com/sql/func_mysql_upper.asp

> Alters single document

```sql
SELECT UPPER(CustomerName) AS UppercaseCustomerName
```

### ABS
https://www.w3schools.com/sql/func_mysql_abs.asp

> Alters single document

```sql
SELECT ABS(-243.5);
```

### AVG
https://www.w3schools.com/sql/func_mysql_avg.asp

> Limits the result to 1

```sql
SELECT AVG(Price) AS AveragePrice
```

### MAX
https://www.w3schools.com/sql/func_mysql_max.asp

> Limits result to 1

```sql
SELECT MAX(Price) AS LargestPrice
```

### MIN
https://www.w3schools.com/sql/func_mysql_min.asp

> Limits result to 1

```sql
SELECT MIN(Price) AS SmallestPrice
```

### ROUND
https://www.w3schools.com/sql/func_mysql_round.asp

> Alters single document

```sql
SELECT ProductName, Price, ROUND(Price, 1) AS RoundedPrice
```

### SUM
https://www.w3schools.com/sql/func_mysql_sum.asp

> Limits result to 1

```sql
SELECT SUM(Quantity) AS TotalItemsOrdered
```

**Comment by REDAL** [comment]
> #169 brought the idea.
> 
> We could implement functions for select (like we have with select in conjunction with groupBy).
> 
> # [List of all Mysql-Functions](https://www.w3schools.com/sql/sql_ref_mysql.asp)
> I think the following functions could be useful for us? Would like to hear your opinions!
> 
> ## String functions
> ### CONCAT
> https://www.w3schools.com/sql/func_mysql_concat_ws.asp
> 
> ```sql
> SELECT CONCAT_WS(" ", Address, PostalCode, City) AS Address
> ```
> 
> ### LENGTH
> https://www.w3schools.com/sql/func_mysql_length.asp
> 
> ```sql
> SELECT LENGTH(CustomerName) AS LengthOfName
> ```
> 
> ### LOWER
> https://www.w3schools.com/sql/func_mysql_lower.asp
> 
> ```sql
> SELECT LOWER(CustomerName) AS LowercaseCustomerName
> ```
> 
> ### POSITION
> https://www.w3schools.com/sql/func_mysql_position.asp
> 
> OR `LOCATE` OR `INSTR`
> They do all the same thing, but I am not sure what's the best naming.
> 
> ```sql
> SELECT POSITION("a" IN CustomerName)
> ```
> 
> ### REPEAT
> https://www.w3schools.com/sql/func_mysql_repeat.asp
> 
> ```sql
> SELECT REPEAT(CustomerName, 2)
> ```
> 
> ### UPPER
> https://www.w3schools.com/sql/func_mysql_upper.asp
> 
> ```sql
> SELECT UPPER(CustomerName) AS UppercaseCustomerName
> ```
> 
> ## Numeric functions
> ### ABS
> https://www.w3schools.com/sql/func_mysql_abs.asp
> 
> ```sql
> SELECT ABS(-243.5);
> ```
> 
> ### AVG
> https://www.w3schools.com/sql/func_mysql_avg.asp
> 
> Limits the result to 1
> 
> ```sql
> SELECT AVG(Price) AS AveragePrice
> ```
> 
> ### MAX
> https://www.w3schools.com/sql/func_mysql_max.asp
> 
> Limits result to 1
> 
> ```sql
> SELECT MAX(Price) AS LargestPrice
> ```
> 
> ### MIN
> https://www.w3schools.com/sql/func_mysql_min.asp
> 
> Limits result to 1
> 
> ```sql
> SELECT MIN(Price) AS SmallestPrice
> ```
> 
> ### ROUND
> https://www.w3schools.com/sql/func_mysql_round.asp
> 
> ```sql
> SELECT ProductName, Price, ROUND(Price, 1) AS RoundedPrice
> ```
> 
> ### SUM
> https://www.w3schools.com/sql/func_mysql_sum.asp
> 
> Limits result to 1
> 
> ```sql
> SELECT SUM(Quantity) AS TotalItemsOrdered
> ```

awesome, like I said before, SleekDB is moving forward to have like functions as MySQL :) why not 

I APPROVE

**Comment by Timu57** [comment]
Implemented with version 2.9
